### PR TITLE
Structure-highlighting reformat of the test for issue 8368

### PR DESCRIPTION
**Pull request by toolforger**
This is in preparation for a PR that will need to replace a condition
in a Piecewise.
Improvement proposals happily accepted.

### CODE REVIEW

File: sympy/integrals/tests/test_integrals.py [hunk 1]
```diff
@@ -1070,17 +1070,36 @@ def test_issue_2708():
...
 def test_issue_8368():
     assert integrate(exp(-s*x)*cosh(x), (x, 0, oo)) == \
-        Piecewise((pi*Piecewise((-s/(pi*(-s**2 + 1)), Abs(s**2) < 1),
-        (1/(pi*s*(1 - 1/s**2)), Abs(s**(-2)) < 1), (meijerg(((S(1)/2,), (0, 0)),
-        ((0, S(1)/2), (0,)), polar_lift(s)**2), True)),
-        And(Abs(periodic_argument(polar_lift(s)**2, oo)) < pi,
-        cos(Abs(periodic_argument(polar_lift(s)**2, oo))/2)*sqrt(Abs(s**2)) -
-        1 > 0, Ne(s**2, 1))), (Integral(exp(-s*x)*cosh(x), (x, 0, oo)), True))
+        Piecewise(
+            (   pi*Piecewise(
```
Comment (jcrist) [inline sympy/integrals/tests/test_integrals.py]: Ahhhhhh, spaces in brackets make my eyes bleed:/ The indent formatting makes the flow of piecewise easier to read, but the spaces should be removed. Why did you add them?

**Comment by jcrist** [comment]
Besides my picky formatting comment, I have no objections. Tests should be readable, this helps significantly.

File: sympy/integrals/tests/test_integrals.py (hunk 1 above)
Comment (toolforger) [inline sympy/integrals/tests/test_integrals.py]: LOL understood

Here's how I arrived at that:

I was going for

```
function(
    subexpression1,
    subexpression2
)
```

Now for tuples, I do not have a function name, which gave me

```
(
    subexpression1,
    subexpression2
)
```

which looks silly and wastes an entire line for the merely structural opening brace. So I folded the first line back but had to keep the subexpressions aligned, which gave me

```
(   subexpression1,
    subexpression2
)
```

I'm not really happy with that myself; better proposals how to format multi-line tuples very much appreciated.

**Comment by toolforger** [comment]
I suppose this test was created by running the code, checking the result is mathematically correct, and then doing copy&paste from `str(expression)` to code.
Which is okay, except the test cannot be checked for validity and is fragile against changes in the various heuristics in SymPy.

I guess the real problem is that `Integrate` takes a tuple as its second argument. This adds an additional level of nesting, which is really bad for readability.
`Piecewise` has a similar problem; I see the motivation for grouping, but it's cluttering expressions.
I am wondering whether I should open an Issue for this.

**Comment by toolforger** [comment]
ping @jcrist @asmeurer @certik 
This test will have to be modified in my upcoming `ordering_of_classes` PR. This PR helps making that future PR reviewable.

File: sympy/integrals/tests/test_integrals.py (hunk 1 above)
Comment (jcrist) [inline sympy/integrals/tests/test_integrals.py]: The formatting issue still stands. Perhaps just dedenting all the text one level back (or the parens one level forward) would make it more readable, and still not have spaces in parens? Something like:

```
(sub expression,
    (subexpression1,
     subexpression2),
 otherstuff)...
```
idk. I'd be ok with whatever seems reable to you, but has *no spaces* between the parens and the actual expression.

Comment (toolforger) [inline sympy/integrals/tests/test_integrals.py]: I'd want stuff that at's the same level to be indented to the same level, so

```
(subexpression
    (subexpression
```

does not work very well. Issue here is that the opening parentheses belongs at the same level as `subexpression` but `subexpression1` belongs one level further.
What I wouldn't like but find acceptable:

```
(subexpression,
(
    subexpression1,
    subexpression2),
otherstuff)
```

It's looking weird, but I guess everything will look weird.

Another (and usually better) way out is to extract subexpressions out into local variables. It does not solve all such problems but can reduce them. Something like

```
piecewise11 = (-s/(pi*(-s**2 + 1)), Abs(s**2) < 1)
piecewise12 = (1/(pi*s*(1 - 1/s**2)), Abs(s**(-2)) < 1)
piecewise13 = (meijerg(((S(1)/2,), (0, 0)), ((0, S(1)/2), (0,)), polar_lift(s)**2), True)
condition11 = Abs(periodic_argument(polar_lift(s)**2, oo)) < pi
condition12 = cos(Abs(periodic_argument(polar_lift(s)**2, oo))/2)*sqrt(Abs(s**2)) - 1 > 0,
condition13 = Ne(s**2, 1)
piecewise1 = ((piecewise11, piecewise12, piecewise13), condition1)
condition1 = And(condition11, condition12, condition13)
piecewise2 = (Integral(exp(-s*x)*cosh(x), (x, 0, oo)), True)
assert Piecewise((piecewise1, condition1), piecewise2)
```

Just to demonstrate the general approach, I think there are some typos in that, and I'm not 100% content with the naming yet.

**Comment by certik** [comment]
Thanks for improving the test. I agree that the spaces around ( are weird, but that's easy to fix, so we can do it later after we agree upon a style.
